go/chat/sender_test.go

```diff
@@ -1042,14 +1047,8 @@ func TestPrevPointerAddition(t *testing.T) {
 		if ephemeralLifetime != nil {
 			t.Logf("expiry all ephemeral messages")
 			world.Fc.Advance(time.Second*time.Duration(*ephemeralLifetime) + chat1.ShowExplosionLifetime)
-			// Mock out server call for new messages
-			ri.GetThreadRemoteFunc = func(m *kbtest.ChatRemoteMock, ctx context.Context, arg chat1.GetThreadRemoteArg) (chat1.GetThreadRemoteRes, error) {
-				return chat1.GetThreadRemoteRes{
-					Thread: chat1.ThreadViewBoxed{
-						Pagination: &chat1.Pagination{},
-					},
-				}, nil
-			}
+			// Mock out pulling messages to return no messages
```

Comment:
Actually why did you do this? I don't like all that random code in convsource

Reply:
This test depends on `ConvSource` returning zero messages from a `Pull` (line 1068). Previously mocking `GetThreadRemoteFunc` seemed to be enough. Note that the pagination number is set to 2 for this test (line 197). 

This patch changes the definition of pagination. Meaning that even with the count set to 2 _any fetch_ will fill in the whole conv in cache in this test since a bunch of messages are deleted, and so do not count towards the paging limit. In addition to whatever caching mechanisms there are background processes that fill in cache. Those background processes would have to be shut off for this test to work. The background processes include the ephemeral doohicky, and idk whether BackgroundLoader is running or not. But it's easier just to mock out `Pull`. It didn't seem too far fetched since the existing mocking out of `GetThreadRemoteFunc` was not real behavior anyway.

cc @joshblum

Reply:
Ok at least stick it in `postProcessThread` then, so that stuff is only in one place. I just dislike having too much `if <I am testing>` code hanging around.

Reply:
Can do. What do you mean in one place? Where's the `if <I am testing>` in [`postProcessThread`](https://github.com/keybase/client/blob/master/go/chat/convsource.go#L73)?

Reply:
Oh I see that it's the common callee